agent/util/archive_tar_test.go

```diff
@@ -5,26 +5,40 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"runtime"
 	"strings"
 	"testing"
 
 	"github.com/evergreen-ci/utility"
 	"github.com/mongodb/grip/logging"
 	. "github.com/smartystreets/goconvey/convey"
 	"github.com/smartystreets/goconvey/convey/reporting"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 func init() {
 	reporting.QuietMode()
 }
 
+func getDirectoryOfFile() string {
+	// returns the directory of the file of the calling
+	// function. duplicated from testutil to avoid a cycle.
+	_, file, _, _ := runtime.Caller(1)
+
+	return filepath.Dir(file)
+}
+
 func TestArchiveExtract(t *testing.T) {
 	Convey("After extracting a tarball", t, func() {
 		testDir := getDirectoryOfFile()
 		//Remove the test output dir, in case it was left over from prior test
-		err := os.RemoveAll(filepath.Join(testDir, "testdata", "artifacts_test"))
+		outputDir := filepath.Join(testDir, "testdata", "artifacts_test")
```

Comment:
Can we just use a single temp dir for all this so we can just blow it up at the end and not worry about removing all these individual files and sub directories?

Reply:
I changed them to use `ioutil.TempDir`. We can't share the temp dirs because the tests are verifying that the exact contents of the directory look like what's expected from untarring.